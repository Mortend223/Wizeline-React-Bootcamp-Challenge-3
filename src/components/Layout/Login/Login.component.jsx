import React, { useState } from "react";

// Styles
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faTimesCircle } from "@fortawesome/free-solid-svg-icons";
import { Button, ErrorMessage, Section } from "./Login.styles";

// Providers
import { useAuth } from "../../../providers/Auth/Auth.provider";

function Login() {
  const { login, loading, error } = useAuth();
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");

  const handleLogin = async () => {
    await login(email, password);
  };
  return (
    <Section>
      {error && (
        <ErrorMessage className="error-msg">
          <FontAwesomeIcon
            icon={faTimesCircle}
            size="1x"
            style={{ color: "red" }}
            title="error-message"
          />
          {error}
        </ErrorMessage>
      )}
      <div className="form-group">
        <label htmlFor="email">
          <strong>Email </strong>
          <input
            required
            type="email"
            id="email"
            onChange={(e) => setEmail(e.target.value)}
          />
        </label>
      </div>
      <div className="form-group">
        <label htmlFor="password">
          <strong>Password </strong>
          <input
            required
            type="password"
            id="password"
            onChange={(e) => setPassword(e.target.value)}
          />
        </label>
      </div>
      <Button
        type="button"
        onClick={handleLogin}
        color="primary"
        disabled={loading}
      >
        {loading ? "Login in..." : "Login"}
      </Button>
    </Section>
  );
}

export default Login;
