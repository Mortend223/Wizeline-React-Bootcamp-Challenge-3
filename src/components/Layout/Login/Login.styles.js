import styled from "styled-components";

export const Button = styled.button`
  background-color: #f0575d;
  border: 1px solid #f0575d;
  border-radius: 10px;
  color: white;
  cursor: pointer;
  display: block;
  font-size: 16px;
  margin-left: auto;
  margin-right: 0px;
  padding: 15px 32px;
  text-align: center;
  text-decoration: none;
  width: 48%;
  &:hover {
    background-color: #fff;
    border: 1px solid black;
    color: black;
  }
`;

export const ErrorMessage = styled.div`
  background-color: #ffbaba;
  border-radius: 10px;
  color: #d8000c;
  font-size: 16px;
  margin: 10px 0;
  padding: 10px;
  svg {
    margin-right: 5px;
  }
`;

export const Section = styled.div`
  border: 1px solid black;
  border-radius: 10px;
  font-family: "Roboto", "Helvetica", "Arial", sans-serif;
  font-size: 1.5rem;
  max-width: 300px;
  padding: 20px;
  margin: auto;
  text-align: left;
  label {
    display: block;
  }
  input {
    border-radius: 10px;
    display: block;
    font-size: 14px;
    height: 30px;
    margin-bottom: 20px;
    padding: 5px;
    width: 100%;
  }
`;
