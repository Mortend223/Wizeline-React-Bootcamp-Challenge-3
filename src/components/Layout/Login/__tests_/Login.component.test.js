import React from "react";
import { fireEvent, render, screen } from "@testing-library/react";

// Components
import Login from "../Login.component";

// Providers
import { useAuth } from "../../../../providers/Auth/Auth.provider";

jest.mock("../../../../providers/Auth/Auth.provider");

describe("Login Component", () => {
  it("Should show Login Component", () => {
    useAuth.mockReturnValue({
      authenticated: true,
      login: jest.fn(),
      user: {},
    });
    const container = render(<Login onClick={jest.fn()} />);

    expect(container.firstChild).toMatchSnapshot();
  });
  it("Should show Login Component with Loading button", () => {
    useAuth.mockReturnValue({
      authenticated: true,
      login: jest.fn(),
      loading: true,
    });
    const container = render(<Login />);

    expect(container.firstChild).toMatchSnapshot();
  });
  it("Should show Login Component with Error", () => {
    useAuth.mockReturnValue({
      error: true,
      login: jest.fn(),
      loading: false,
    });
    const container = render(<Login />);

    expect(container.firstChild).toMatchSnapshot();
  });
  it("Should show Email Change", () => {
    useAuth.mockReturnValue({
      authenticated: true,
      login: jest.fn(),
      user: {},
    });
    const query = "SomeText";
    render(<Login onClick={jest.fn()} />);

    const emailInput = screen.getByLabelText(/email/i);
    expect(emailInput.value).toEqual("");
    fireEvent.change(emailInput, { target: { value: query } });
    expect(emailInput.value).toEqual(query);
  });
  it("Should show Pass Change", () => {
    useAuth.mockReturnValue({
      authenticated: true,
      login: jest.fn(),
      user: {},
    });
    const query = "SomeText";
    const container = render(<Login onClick={jest.fn()} />);

    expect(container.firstChild).toMatchSnapshot();
    const passInput = screen.getByLabelText(/Password/i);
    expect(passInput.value).toEqual("");
    fireEvent.change(passInput, { target: { value: query } });
    expect(passInput.value).toEqual(query);
  });
});
