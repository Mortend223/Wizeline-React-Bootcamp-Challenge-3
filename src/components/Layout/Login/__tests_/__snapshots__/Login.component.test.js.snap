// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component Should show Login Component 1`] = `undefined`;

exports[`Login Component Should show Login Component with Error 1`] = `undefined`;

exports[`Login Component Should show Login Component with Loading button 1`] = `undefined`;

exports[`Login Component Should show Pass Change 1`] = `undefined`;
